File: PHP/Medium/Sorting/1561__Maximum_Number_of_Coins_You_Can_Get.php
<?php
//function maxCoins($piles)
//{
//    rsort($piles);
//    $count = 0;
//    while (!empty($piles)) {
//        array_shift($piles);
//        $count += array_shift($piles);
//        array_pop($piles);
//    }
//    return $count;
//}

function maxCoins($piles)
{
    rsort($piles);
    $count = 0;
    $n = count($piles) / 3;

    for ($i = 0; $i < $n; $i++) {
        $count += $piles[$i * 2 + 1];
    }

    return $count;
}

$piles = [2, 4, 1, 2, 7, 8];
var_dump(maxCoins($piles));

File: Python/Medium/Sorting/1561__Maximum_Number_of_Coins_You_Can_Get.py
def maxCoins(piles: list[int]) -> int:
    # piles.sort(reverse=True)
    # count = 0
    #
    # while piles:
    #     piles.pop(0)
    #     count += piles.pop(0)
    #     piles.pop(-1)
    # return count
# 2 method
    piles.sort(reverse=True)
    count = 0
    n = len(piles) // 3
    for i in range(n):
        count += piles[i * 2 + 1]
    return count


piles = [2, 4, 5]
# piles = [2, 4, 1, 2, 7, 8]
# piles = [9, 8, 7, 6, 5, 1, 2, 3, 4]
print(maxCoins(piles))
